libs/api/utils/src/lib/api-utils.ts

```diff
@@ -20,11 +44,61 @@ export async function validateApiKey(apiKey: string): Promise<ApiErrorCode | und
   if (key.is_deleted) return 'PUBLIC_API_KEY_HAS_BEEN_DELETED'
   if (!key.is_active) return 'PUBLIC_API_KEY_IS_NOT_ACTIVE'
   if (isDateBeforeNow(key.expiration_date)) return 'PUBLIC_API_KEY_HAS_BEEN_EXPIRED'
+  API_KEY_TO_ID_CACHE[apiKey] = key.id
+  API_KEY_CONTEXT = apiKey
 
   await incrementKeyUsage(key.id)
   return
 }
 
+export async function resolveApiKeyIdByKey(apiKey: string): Promise<bigint> {
+  const keyId: bigint | undefined = API_KEY_TO_ID_CACHE[apiKey as keyof typeof API_KEY_TO_ID_CACHE]
```

Comment:
**question:** Why can't we just type `apiKey` as keyof typeof API_KEY_TO_ID_CACHE directly in function definition instead of string and then casting it as this type?